### PR TITLE
Personal Circumstance for offender

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/data/api/PersonalCircumstance.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import lombok.Data;
+
+import java.time.LocalDateTime;
+
+@Data
+@Builder
+public class PersonalCircumstance {
+    private Long personalCircumstanceId;
+    private Long offenderId;
+    private KeyValue type;
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/data/api/PersonalCircumstance.java]: Should these be called `type` and `subType` or `personalCircumstanceType` etc ?

File: src/main/java/uk/gov/justice/digital/delius/service/PersonalCircumstanceService.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        this.personalCircumstanceRepository = personalCircumstanceRepository;
+        this.personalCircumstanceTransformer = personalCircumstanceTransformer;
+    }
+
+    public List<PersonalCircumstance> personalCircumstancesFor(Long offenderId) {
+        List<uk.gov.justice.digital.delius.jpa.standard.entity.PersonalCircumstance> personalCircumstances = personalCircumstanceRepository.findByOffenderId(offenderId);
+        return personalCircumstances
+                .stream()
+                .filter(personalCircumstance -> !convertToBoolean(personalCircumstance.getSoftDeleted()))
+                .sorted(Comparator.comparing(uk.gov.justice.digital.delius.jpa.standard.entity.PersonalCircumstance::getPersonalCircumstanceId).reversed())
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/service/PersonalCircumstanceService.java]: There is no sensible order, so just ordering be insertion reversed. Any one have any better ideas? Delius orders on the UI by `startDate`

File: src/main/resources/db/migration/V1_12__schema.sql [hunk 3]
```diff
@@ -0,0 +1,142 @@
+CREATE TABLE R_CIRCUMSTANCE_SUB_TYPE
```
Comment (andymarke) [inline src/main/resources/db/migration/V1_12__schema.sql]: Back fill some tables that we map but were missing for DEV

File: src/main/java/uk/gov/justice/digital/delius/data/api/PersonalCircumstance.java (hunk 1 above)
Comment (llmikeyj) [reply src/main/java/uk/gov/justice/digital/delius/data/api/PersonalCircumstance.java]: It's a toughie with anonymous tuples like this. My gut feeling would be to call it 'value' for want of a better vague term

File: src/main/java/uk/gov/justice/digital/delius/service/PersonalCircumstanceService.java (hunk 2 above)
Comment (llmikeyj) [reply src/main/java/uk/gov/justice/digital/delius/service/PersonalCircumstanceService.java]: Unless you understand the subject of 'more important' Personal Circumstances, I think that's about all we can do unless there's useful timestamps maybe

File: src/main/java/uk/gov/justice/digital/delius/data/api/PersonalCircumstance.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    private KeyValue subType;
+    private LocalDateTime startDate;
```
Comment (llmikeyj) [inline src/main/java/uk/gov/justice/digital/delius/data/api/PersonalCircumstance.java]: Do these need to be date times? Or would date suffice?

File: src/main/java/uk/gov/justice/digital/delius/transformers/PersonalCircumstanceTransformer.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                .probationArea(Optional.ofNullable(personalCircumstance.getProbationArea()).map(this::probationAreaOf).orElse(null))
+                .endDate(personalCircumstance.getEndDate())
+                .startDate(personalCircumstance.getStartDate())
+                .offenderId(personalCircumstance.getOffenderId())
+                .notes(personalCircumstance.getNotes())
+                .evidenced(ynToBoolean(personalCircumstance.getEvidenced()))
+            .build();
+    }
+
+    private KeyValue probationAreaOf(ProbationArea probationArea) {
```
Comment (llmikeyj) [inline src/main/java/uk/gov/justice/digital/delius/transformers/PersonalCircumstanceTransformer.java]: This transform probably already exists in another transformer

Comment (andymarke) [reply src/main/java/uk/gov/justice/digital/delius/transformers/PersonalCircumstanceTransformer.java]: The key/value pair version is duplicated in a few transformers, perhaps we should add  a `public KeyValue keyValueOf(ProbabtionArea probationArea)` to the ProbationAreaTransformer` along side the existing `probationAreaOf` used by the ReferenceData service? Feels a bit OTT though

File: src/main/java/uk/gov/justice/digital/delius/data/api/PersonalCircumstance.java (hunk 1 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/digital/delius/data/api/PersonalCircumstance.java]: yeah, I found a precedence with `Nsi` model, which calls it `nsiType` and `nsiSubType` - so renamed to be consistent.

**Pull request merged**